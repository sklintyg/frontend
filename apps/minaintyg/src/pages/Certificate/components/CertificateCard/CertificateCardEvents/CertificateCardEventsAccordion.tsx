import { IDSIconMinus, IDSIconPlus } from '@inera/ids-react'
import type { ReactNode } from 'react'

export function CertificateCardEventsAccordion({ children }: { children: ReactNode }) {
  return (
    <details className="group py-2.5">
      <summary className="flex items-center justify-between gap-2 font-semibold text-accent-40 group-open:mb-2.5 md:hidden">
        <h4 className="inline-block group-open:hidden">Visa händelser</h4>
        <h4 className="hidden group-open:inline-block">Dölj händelser</h4>
        <IDSIconPlus
          width="100%"
          height="100%"
          inline
          className="h-5 w-5 rounded-full bg-accent-40 p-1 text-white group-open:hidden"
          color="currentColor"
        />
        <IDSIconMinus
          width="100%"
          height="100%"
          inline
          className="hidden h-5 w-5 rounded-full bg-accent-40 p-1 text-white group-open:flex"
          color="currentColor"
        />
      </summary>
      {children}
    </details>
  )
}
