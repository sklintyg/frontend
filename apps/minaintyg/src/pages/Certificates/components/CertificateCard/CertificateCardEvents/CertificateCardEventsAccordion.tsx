import { IDSIconMinus, IDSIconPlus } from '@frontend/ids-react-ts'
import { ReactNode } from 'react'

export function CertificateCardEventsAccordion({ children }: { children: ReactNode }) {
  return (
    <details className="group py-2.5">
      <summary className="flex items-center justify-between gap-2 font-semibold text-sky-base group-open:mb-2.5 md:hidden">
        <span className="inline-block group-open:hidden">Visa händelser</span>
        <span className="hidden group-open:inline-block">Dölj händelser</span>
        <IDSIconPlus
          width="100%"
          height="100%"
          inline
          className="h-5 w-5 rounded-full bg-sky-base p-1 text-white group-open:hidden"
          color="currentColor"
        />
        <IDSIconMinus
          width="100%"
          height="100%"
          inline
          className="hidden h-5 w-5 rounded-full bg-sky-base p-1 text-white group-open:flex"
          color="currentColor"
        />
      </summary>
      {children}
    </details>
  )
}
