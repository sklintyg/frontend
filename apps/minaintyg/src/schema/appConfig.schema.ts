import { z } from 'zod'

export const appConfigLinkSchema = z.object({
  key: z.string(),
  url: z.string(),
  text: z.string(),
  target: z.string(),
})

export const appConfigSchema = z.object({
  useMinifiedJavascript: z.boolean(),
  version: z.string(),
  buildNumber: z.string(),
  elva77MainUrl: z.string(),
  elva77LoginUrl: z.string(),
  applicationLogoutUrl: z.string(),
  miUserSurveyUrl: z.string(),
  miUserSurveyVersion: z.string(),
  miUserSurveyDateFrom: z.string(),
  miUserSurveyDateTo: z.string(),
  knownRecipients: z.array(z.object({ id: z.string(), name: z.string(), trusted: z.boolean() })),
  links: z.record(appConfigLinkSchema),
  banners: z.array(z.unknown()),
})

export type AppConfig = z.infer<typeof appConfigSchema>
export type AppConfigLink = z.infer<typeof appConfigLinkSchema>
