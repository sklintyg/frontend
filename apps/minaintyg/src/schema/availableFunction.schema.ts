import { z } from 'zod'

export const AvailableFunctionType = z.enum(['SEND_CERTIFICATE'])
export const AvailableFunctionInformationType = z.enum([''])

export const availableFunctionInformationSchema = z.object({
  type: AvailableFunctionInformationType,
  id: z.string(),
})

export const availableFunctionSchema = z.object({
  body: z.string().nullable(),
  description: z.string().nullable(),
  name: z.string().nullable(),
  title: z.string().nullable(),
  type: AvailableFunctionType,
  information: z.array(availableFunctionInformationSchema),
})

export type AvailableFunction = z.infer<typeof availableFunctionSchema>
