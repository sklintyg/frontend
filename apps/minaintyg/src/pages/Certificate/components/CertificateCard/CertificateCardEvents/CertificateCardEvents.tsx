import { ReactNode } from 'react'
import { CertificateEvent } from '../../../../../schema/certificate.schema'
import { CertificateEventsInfo } from '../../CertificateEventsInfo/CertificateEventsInfo'
import { CertificateCardEventsAccordion } from './CertificateCardEventsAccordion'

export function CertificateCardEvents({ events, heading }: { events: CertificateEvent[]; heading: ReactNode }) {
  return (
    <>
      <div className="md:hidden">
        <CertificateCardEventsAccordion>
          {heading}
          <CertificateEventsInfo events={events} />
        </CertificateCardEventsAccordion>
      </div>
      <div className="hidden md:block">
        {heading}
        <CertificateEventsInfo events={events} />
      </div>
    </>
  )
}
