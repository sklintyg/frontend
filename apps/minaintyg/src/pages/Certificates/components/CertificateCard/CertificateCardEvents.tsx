import { useFormat } from '../../../../hooks/useFormat'
import { CertificateListEvent } from '../../../../schema/certificateList.schema'

export function CertificateCardEvents({ events }: { events: CertificateListEvent[] }) {
  const { formatDatetime } = useFormat()
  return (
    <div>
      <div className="font-bold">
        Senaste händelser (visar {events.length} av {events.length} händelser)
      </div>
      {events.map(({ timestamp, description }) => (
        <div key={timestamp}>
          {formatDatetime(timestamp)} {description}
        </div>
      ))}
    </div>
  )
}
