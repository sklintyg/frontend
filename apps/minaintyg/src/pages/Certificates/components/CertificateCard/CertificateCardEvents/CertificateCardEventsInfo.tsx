import { useFormat } from '../../../../../hooks/useFormat'
import { CertificateListEvent } from '../../../../../schema/certificateList.schema'

export function CertificateCardEventsInfo({ events }: { events: CertificateListEvent[] }) {
  const { datetime } = useFormat()
  return (
    <div>
      <div className="font-bold">
        Senaste händelser (visar {events.length} av {events.length} händelser)
      </div>
      {events.map(({ timestamp, description }) => (
        <div key={timestamp}>
          {datetime(timestamp)} {description}
        </div>
      ))}
    </div>
  )
}
