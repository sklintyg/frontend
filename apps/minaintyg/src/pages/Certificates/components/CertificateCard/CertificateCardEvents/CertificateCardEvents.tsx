import { CertificateListEvent } from '../../../../../schema/certificateList.schema'
import { CertificateCardEventsAccordion } from './CertificateCardEventsAccordion'
import { CertificateCardEventsInfo } from './CertificateCardEventsInfo'

export function CertificateCardEvents({ events }: { events: CertificateListEvent[] }) {
  return (
    <>
      <div className="md:hidden">
        <CertificateCardEventsAccordion>
          <CertificateCardEventsInfo events={events} />
        </CertificateCardEventsAccordion>
      </div>
      <div className="hidden md:block">
        <CertificateCardEventsInfo events={events} />
      </div>
    </>
  )
}
