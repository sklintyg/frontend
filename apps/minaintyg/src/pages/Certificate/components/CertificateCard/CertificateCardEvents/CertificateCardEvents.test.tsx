import { fakerFromSchema } from '@frontend/fake'
import { render } from '@testing-library/react'
import { certificateEventSchema } from '../../../../../schema/certificate.schema'
import { CertificateCardEvents } from './CertificateCardEvents'

it('Should render correctly', () => {
  const events = [
    fakerFromSchema(certificateEventSchema, { seed: 1 })({ timestamp: '2023-09-06T11:00:00.000Z' }),
    fakerFromSchema(certificateEventSchema, { seed: 2 })({ timestamp: '2023-09-05T11:00:00.000Z' }),
    fakerFromSchema(certificateEventSchema, { seed: 3 })({ timestamp: '2023-09-04T11:00:00.000Z' }),
  ]
  const { container } = render(<CertificateCardEvents events={events} />)
  expect(container).toMatchSnapshot()
})
