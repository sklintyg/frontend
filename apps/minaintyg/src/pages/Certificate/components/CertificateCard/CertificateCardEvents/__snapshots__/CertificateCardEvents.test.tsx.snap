// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render correctly 1`] = `
<div>
  <div
    class="md:hidden"
  >
    <details
      class="group py-2.5"
    >
      <summary
        class="flex items-center justify-between gap-2 font-semibold text-sky-base group-open:mb-2.5 md:hidden"
      >
        <span
          class="inline-block group-open:hidden"
        >
          Visa händelser
        </span>
        <span
          class="hidden group-open:inline-block"
        >
          Dölj händelser
        </span>
        <ids-icon-plus
          class="h-5 w-5 rounded-full bg-sky-base p-1 text-white group-open:hidden ids-icon"
        />
        <ids-icon-minus
          class="hidden h-5 w-5 rounded-full bg-sky-base p-1 text-white group-open:flex ids-icon"
        />
      </summary>
      <div>
        <div
          class="font-bold"
        >
          Senaste händelser
        </div>
        <div
          class="flex flex-col md:flex-row md:gap-1"
        >
          <span
            class="whitespace-nowrap"
          >
            2023-09-06, 11:00
            :
          </span>
          <span>
            excepturi
          </span>
        </div>
        <div
          class="flex flex-col md:flex-row md:gap-1"
        >
          <span
            class="whitespace-nowrap"
          >
            2023-09-05, 11:00
            :
          </span>
          <span>
            pariatur
          </span>
        </div>
        <div
          class="flex flex-col md:flex-row md:gap-1"
        >
          <span
            class="whitespace-nowrap"
          >
            2023-09-04, 11:00
            :
          </span>
          <span>
            provident
          </span>
        </div>
      </div>
    </details>
  </div>
  <div
    class="hidden md:block"
  >
    <div>
      <div
        class="font-bold"
      >
        Senaste händelser
      </div>
      <div
        class="flex flex-col md:flex-row md:gap-1"
      >
        <span
          class="whitespace-nowrap"
        >
          2023-09-06, 11:00
          :
        </span>
        <span>
          excepturi
        </span>
      </div>
      <div
        class="flex flex-col md:flex-row md:gap-1"
      >
        <span
          class="whitespace-nowrap"
        >
          2023-09-05, 11:00
          :
        </span>
        <span>
          pariatur
        </span>
      </div>
      <div
        class="flex flex-col md:flex-row md:gap-1"
      >
        <span
          class="whitespace-nowrap"
        >
          2023-09-04, 11:00
          :
        </span>
        <span>
          provident
        </span>
      </div>
    </div>
  </div>
</div>
`;
